controllers/ibmpowervscluster_controller_test.go

```diff
@@ -456,3 +465,174 @@ func cleanupCluster(g *WithT, powervsCluster *infrav1beta2.IBMPowerVSCluster, na
 		}(powervsCluster, namespace)
 	}
 }
+
+func TestReconcilePowerVSResources(t *testing.T) {
+	testCases := []struct {
+		name                    string
+		powerVSClusterScopeFunc func() *scope.PowerVSClusterScope
+		reconcileResult         reconcileResult
+		conditions              capiv1beta1.Conditions
+	}{
+		{
+			name: "when Reconciling PowerVS service instance returns error",
+			powerVSClusterScopeFunc: func() *scope.PowerVSClusterScope {
+				clusterScope := &scope.PowerVSClusterScope{
+					IBMPowerVSCluster: &infrav1beta2.IBMPowerVSCluster{
+						Status: infrav1beta2.IBMPowerVSClusterStatus{
+							ServiceInstance: &infrav1beta2.ResourceReference{
+								ID: ptr.To("serviceInstanceID"),
+							},
+						},
+					},
+				}
+				mockResourceController := mockRC.NewMockResourceController(gomock.NewController(t))
+				mockResourceController.EXPECT().GetResourceInstance(gomock.Any()).Return(nil, nil, fmt.Errorf("error getting resource instance"))
+				clusterScope.ResourceClient = mockResourceController
+				return clusterScope
+			},
+			reconcileResult: reconcileResult{
+				error: errors.New("error getting resource instance"),
+			},
+
+			conditions: capiv1beta1.Conditions{
+				capiv1beta1.Condition{
+					Type:               infrav1beta2.ServiceInstanceReadyCondition,
+					Status:             "False",
+					Severity:           capiv1beta1.ConditionSeverityError,
+					LastTransitionTime: metav1.Time{},
+					Reason:             infrav1beta2.ServiceInstanceReconciliationFailedReason,
+					Message:            "error getting resource instance",
+				},
+			},
+		},
+		{
+			name: "when Reconciling PowerVS service instance returns requeue",
```

Comment:
```suggestion
			name: "when Reconciling PowerVS service instance returns requeue as  true",
```
Please change wherever applicable